internal/cometd/client.go

```diff
@@ -0,0 +1,255 @@
+// Copyright © 2022 Meroxa, Inc. and Miquido
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cometd
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/cookiejar"
+
+	"github.com/miquido/conduit-connector-salesforce/internal/cometd/requests"
+	"github.com/miquido/conduit-connector-salesforce/internal/cometd/responses"
+	"github.com/miquido/conduit-connector-salesforce/internal/utils"
+	"golang.org/x/net/publicsuffix"
+)
+
+func NewClient(baseURL, accessToken string) (*Client, error) {
+	jar, err := cookiejar.New(&cookiejar.Options{
+		PublicSuffixList: publicsuffix.List,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		baseURL:        baseURL,
+		accessToken:    accessToken,
+		longPollClient: &http.Client{Jar: jar},
+	}, nil
+}
+
+type Client struct {
+	baseURL        string
+	accessToken    string
+	clientID       string
+	longPollClient *http.Client
+}
+
+// Handshake performs a handshake request.
+// See: https://docs.cometd.org/current7/reference/#_bayeux_meta_handshake
+func (s *Client) Handshake(ctx context.Context) (responses.SuccessfulHandshakeResponse, error) {
```

Comment:
Is there a special reason why we simply don't use https://github.com/sigmavirus24/gobayeux ?

Reply:
There is no particular reason why I rejected this. Once I was done with playing with other API examples (some unofficial Go's Salesforce SDKs) I had pretty much everything done what I needed. However, I'm considering this for https://github.com/miquido/conduit-connector-salesforce/pull/1#discussion_r874734157 and https://github.com/miquido/conduit-connector-salesforce/pull/1#discussion_r874689693. Should be easy to replace since the logic is almost the same.

Reply:
I played with this SDK, and unfortunately it does not work with Salesforce. SF does not fully implement CometD. The problem is with subscriptions. CometD accepts request with an array of topics when subscribing and this SDK uses this syntax, whereas SF's implementation does not allow for it. I'm currently subscribing to multiple topics by executing multiple subscription requests.

Reply:
Thanks for checking that!